File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
void fun(TreeNode* root,vector<int>&res){
    // Base case lelo ; agar tree ka root empty hai 
    if(root==nullptr){
        return;
    }

    // PostOrder Tree Traversal : left-->right-->root
    // left Child ko lo 
    fun(root->left,res);

    // Right Child ko lo 
    fun(root->right,res);

    // Root node ko visit karo 
    res.push_back(root->val);
}

    vector<int> postorderTraversal(TreeNode* root) {
        vector<int>res;                    // Resultant array lelo ; shuru mei khali hi hoga 
        
        //recursive Function call kardo 
        fun(root,res);
        
        return res;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void fun(TreeNode root, List<Integer>res){
        // Base case bna lo; agar tree ka root hi khali hai 
        if(root==null){
            return;
        }

        // Postorder Traversal : Left --> Right --> Root
        // Left child ko lo 
        fun(root.left,res);

        // Right chlid ko lo 
        fun(root.right,res);

        
        // Root ko lo 
        res.add(root.val);
    }

    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer>res=new ArrayList<>();

        // Recursive Call kardo 
        fun(root,res);

        return res;
    }
}